tasks/omnibus.py

```diff
@@ -624,3 +626,62 @@ def rpath_edit(ctx, install_path, target_rpath_dd_folder, platform="linux"):
         if install_path in binary_rpath:
             new_rpath = os.path.relpath(target_rpath_dd_folder, os.path.dirname(file))
             _patch_binary_rpath(ctx, new_rpath, install_path, binary_rpath, platform, file)
+
+
+@task
+def deduplicate_files(ctx, directory):
+    # Match: .so, .so.0, .so.0.1, .so.0.1.2, ...
+    LIB_PATTERN = re.compile(r"\.so(?:\.\d+)*$")
```

Comment:
If we'd like to extend it to:
- macOS and Windows shared libs
- Python shared libs
... we may consider extend it to sth like:
```suggestion
    LIB_PATTERN = re.compile(r"\.so(?:\.\d+)*$")
    LIB_PATTERN = re.compile(r"\.(bundle|dll|dylib|pyd|so(?:\.\d+)*)$"
```
Otherwise looks good to me, thanks!